packages/x-date-pickers/src/DatePicker/shared.ts

```diff
@@ -30,40 +52,22 @@ export interface BaseDatePickerProps<TDate>
    */
   openTo?: CalendarPickerView;
   /**
-   * Component that will replace default toolbar renderer.
-   * @default DatePickerToolbar
-   */
-  ToolbarComponent?: React.JSXElementConstructor<BaseToolbarProps<TDate, TDate | null>>;
-  /**
-   * Mobile picker date value placeholder, displaying if `value` === `null`.
-   * @default '–'
-   */
-  toolbarPlaceholder?: React.ReactNode;
-  /**
-   * Date format, that is displaying in toolbar.
+   * Array of views to show.
+   * @default ['year', 'day']
    */
-  toolbarFormat?: string;
+  views?: readonly CalendarPickerView[];
   /**
-   * Mobile picker title, displaying in the toolbar.
-   * @default 'Select date'
+   * Overrideable components.
+   * @default {}
    */
-  toolbarTitle?: React.ReactNode;
+  components?: Partial<BaseDatePickerSlotsComponent<TDate>>;
   /**
-   * Array of views to show.
-   * @default ['year', 'day']
+   * The props used for each component slot.
+   * @default {}
    */
-  views?: readonly CalendarPickerView[];
+  componentsProps?: Partial<BaseDatePickerSlotsComponentsProps<TDate>>;
 }
 
-export const isYearOnlyView = (
```

Comment:
Moved to a utility file to avoid a circular import now that the toolbar is imported in this file (it uses those methods)